### ISSUE TITLE
Message column duplicates if Attributes contains Message.

### ISSUE THREAD

**Issue by vbondaryuk** [dotnet/aspire#3614]
During migration from the old code base, we are going to switch for OTEL. We are trying to use Aspire Dashboard for migration steps as a great toll for visualizing our Logs, Metrics, and traces.

We are goiung to keep our logs in the same format we currently have to keep backward compatibility. Everything works well except logging page. We faced the issue that Message column duplicates as a result filtering stopped to work as well as vilualization.

**Repro steps:**
```cs
using System.Diagnostics.Tracing;
using Microsoft.AspNetCore.Mvc;
using OpenTelemetry.Logs;

var builder = WebApplication.CreateBuilder(args);
builder.Logging.AddOpenTelemetry(logging =>
{
    logging.IncludeFormattedMessage = true;
    logging.IncludeScopes = true; 
});
builder.Services.Configure<OpenTelemetryLoggerOptions>(logging => logging.AddOtlpExporter());

var app = builder.Build();

app.MapGet("/sample", ([FromServices] ILogger<Program> logger) =>
{
    const string message = "Hello World";
    IReadOnlyList<KeyValuePair<string, object?>> eventData = new List<KeyValuePair<string, object?>>()
    {
        new KeyValuePair<string, object?>("Message", message),
        new KeyValuePair<string, object?>("ActivityId", 123),
        new KeyValuePair<string, object?>("Level", (int) EventLevel.Informational),
        new KeyValuePair<string, object?>("Tid", Thread.CurrentThread.ManagedThreadId),
        new KeyValuePair<string, object?>("Pid", Environment.ProcessId),
    };

    logger.Log(LogLevel.Information, 0, eventData, null, formatter: (_, _) => null);
    
    return message;
});
```

**Actual result:**
![image](https://github.com/dotnet/aspire/assets/5966463/1d51ab3f-c822-4621-8777-d8751ebc547c)

**Expected result:**
Only one Message column exist. Message from the Attributes can be renamed or added new Column `Body`.

**Comment by JamesNK** [comment]
Hi

I took a look at this issue, and I don't agree with the recommended fix. It's valid to have an attribute in a log message called `Message`. It shouldn't override the log body message.

For example, `log.LogInformation("Error occurred: {Message}", "Test message");` should result in the UI showing `Error occurred: Test message`. It shouldn't just be `Test message`.

You're seeing this log message:
```
System.Collections.Generic.List`1[System.Collections.Generic.KeyValuePair`2[System.String,System.Object]]
```
because Microsoft.Extensions.Logging calls `ToString()` on the state if there is no formatter provided. You need to change the call to log to include the value you want as the log message body: `logger.Log(LogLevel.Information, 0, eventData, null, formatter: (_, _) => message);`. Or the value passed as state can return it in `ToString()`. This has nothing to do with Aspire dashboard. It's just displaying the information sent to it.

I think there are some improvements here. Filtering for "built in" columns should include attributes with the same name. e.g. Message uses the body message and a message attribute. Also, the dashboard shouldn't show duplicate names in the dropdown.

**Comment by vbondaryuk** [comment]
@JamesNK, yes, we already handled adding formatters. It works, but as a result we have two Message columns.
